x/upgrade/client/cli/query.go

```diff
@@ -1,98 +1,91 @@
 package cli
 
 import (
-	"encoding/binary"
+	"context"
 	"fmt"
 
-	"github.com/cosmos/cosmos-sdk/x/upgrade/types"
-
 	"github.com/spf13/cobra"
 
 	"github.com/cosmos/cosmos-sdk/client"
-	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/x/upgrade/types"
 )
 
-// GetPlanCmd returns the query upgrade plan command
-func GetPlanCmd(storeName string, cdc *codec.Codec) *cobra.Command {
-	return &cobra.Command{
+// GetCurrentPlanCmd returns the query upgrade plan command
+func GetCurrentPlanCmd() *cobra.Command {
+	cmd := &cobra.Command{
 		Use:   "plan",
 		Short: "get upgrade plan (if one exists)",
 		Long:  "Gets the currently scheduled upgrade plan, if one exists",
 		Args:  cobra.ExactArgs(0),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			clientCtx := client.NewContext().WithCodec(cdc).WithJSONMarshaler(cdc)
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			queryClient := types.NewQueryClient(clientCtx.Init())
 
-			// ignore height for now
-			res, _, err := clientCtx.Query(fmt.Sprintf("custom/%s/%s", types.QuerierKey, types.QueryCurrent))
+			params := types.NewQueryCurrentPlanRequest()
+			res, err := queryClient.CurrentPlan(context.Background(), params)
 			if err != nil {
 				return err
 			}
 
-			if len(res) == 0 {
+			if len(res.Plan.Name) == 0 {
 				return fmt.Errorf("no upgrade scheduled")
 			}
 
-			var plan types.Plan
-			err = cdc.UnmarshalJSON(res, &plan)
 			if err != nil {
 				return err
 			}
-			return clientCtx.PrintOutput(plan)
+			return clientCtx.PrintOutput(res.Plan)
```

Comment:
return res?

Reply:
afaiu returning res would return a json with a nested `plan` field like `{"plan": {...}}`, instead of the plan directly. I put `res.Plan` to have the same output as before. Could do both ways.

Reply:
Yes, `res.Plan` makes sense